### PR TITLE
lintr update + pseudoMCMC

### PR DESCRIPTION
**Pull request by robertladwig**
lintr-ized analyse_ncdf.R, run_LHC.R, analzyse_strat.R; 
and added run_MCMC.R for FLake, GLM, GOTM and Simstrat, which enables the user to run a pseudoMCMC algorithm on the data for calibration (alternative to the LHC)

### CODE REVIEW

File: R/analyse_strat.R [hunk 1]
```diff
@@ -24,34 +29,34 @@
...
+          message("Warning: Using ", depths[i], " as the surface.")
         }
         break
       }
     }
     # Find closest depth near the bottom without NA
-    for(i in length(depths):1){
-      Tb = data[data[,2] == depths[i],3]
-      if(sum(is.na(Tb))/length(Tb) < 0.25){
+    for(i in seq_len((length(depths)):1)){
```
Comment (jorritmesman) [inline R/analyse_strat.R]: This will go wrong. Change to rev(seq_len(length(depths))), or leave as is.

File: R/analyse_strat.R [hunk 2]
```diff
@@ -97,34 +106,38 @@ analyse_strat <- function(data = NULL, Ts, Tb, dates, H_ice = NULL, drho = 0.1,
...
+  for(mm in unique(a1$year[!is.na(a1$year)])) { # remove NAs which are generated when the lake is
+    # stratified at the satrt or end of the simulation
+    a2 <- subset(a1, year == mm)
     ind <- which.max(a2$dur)
-    if(nrow(a2)==1) if(is.na(a2$dur)) ind <- NA # fixes issue if stratified at end of data period
-    yr <- c(yr,mm)
-    s.max <- c(s.max,max(a2$dur))
-    s.mean <- c(s.mean,mean(a2$dur))
-    s.tot <- c(s.tot,sum(a2$dur))
+    if(nrow(a2) == 1) if(is.na(a2$dur)) ind <- NA # fixes issue if stratified at end of data period
```
Comment (jorritmesman) [inline R/analyse_strat.R]: This was already in the old code, but I'm not a fan of this if-statement in an if-statement on one line. Feel free to keep if it works, but I'd split it over multiple lines

File: R/calc_stats.R [hunk 3]
```diff
@@ -24,7 +24,7 @@ calc_stats <- function(obs, model, depths, folder = ".", NH, flake_nml, out_time
                                nml_file = flake_nml, long = TRUE,
                                out_time = out_time, out_hour = out_hour)
 
-    stats <- sum_stat(fla_long, obs, depth = TRUE, strat = FALSE)
+    stats <- sum_stat(fla_long, obs, depth = TRUE)
```
Comment (jorritmesman) [inline R/calc_stats.R]: I don't see sum_stat? Is that a file you've added or changed? As long as you're sure this doesn't lead to crashes, fine by me :)

Comment (tadhg-moore) [reply R/calc_stats.R]: sum_stat is from the legendary gotmtools package. Which is 'relatively' stable...

File: R/analyse_strat.R (hunk 2 above)
Comment (robertladwig) [reply R/analyse_strat.R]: This is all legacy code from Tom

File: R/analyse_strat.R (hunk 1 above)
Comment (robertladwig) [reply R/analyse_strat.R]: changed it in an additional commit f5c4766eab1f0bf2ccd7cef2bb04277f52ad8789

File: R/calc_stats.R (hunk 3 above)
Comment (robertladwig) [reply R/calc_stats.R]: changed it to gotmtools::sum_stats 9ba3c13905eb542b4794694b4df276405e5ab681

**Pull request merged**